### PR TITLE
Version 1.8.3

### PR DESCRIPTION
**Pull request by devyoustice**
attempts to work-around many limitations in hosting companies (mainly allow_url_fopen, different php modules). Fix for window hosts, + translations.

### CODE REVIEW

File: views/templates/front/ordersPage.tpl [hunk 1]
```diff
@@ -36,7 +36,9 @@
             {foreach from=$orders item=order}
                 <tr>
                     <td>
-                        <b>{$order->getName()|escape:'htmlall'}</b> ({($order->isPaid() == true) ? 'paid' : 'unpaid'}, {($order->isDelivered() == true) ? 'delivered' : 'undelivered'})<br>
+                        <b>{$order->getName()|escape:'htmlall'}</b>
```
Comment (gaillafr) [inline views/templates/front/ordersPage.tpl]: You should assign some values directly within the controller, views should not be used to treat anything.

File: SDK/ShopItem.php [hunk 2]
```diff
@@ -248,10 +262,12 @@ protected function loadImage($path)
...
 				throw new Exception('Image does not exists');
 
+			$mime_type = $this->getMimeType($path);
+
 			//correct image
 			if (Tools::strlen($image_data) > 0)
 			{
-				$image_data = $this->resize($image_data, 300, 300);
+				$image_data = $this->resize($image_data, 300, 300, false, $mime_type);
 				$a = 'base64_encode';
```
Comment (gaillafr) [inline SDK/ShopItem.php]: If it's only to pass the validation, please don't use this kind of tricks.

**Comment by gaillafr** [comment]
Could you please also update the version in config.xml?
Thank you, kind regards.

**Comment by devyoustice** [comment]
AFAIK, config.xml is auto-generated by presta. The repository does not contain a config.xml file any more.
For the rest, I will make sure the PR is updated.

**Comment by gaillafr** [comment]
Thank you @devyoustice for your reply.
You should let the config.xml file on the repo, it is used by the cache.

Kind regards.

**Comment by devyoustice** [comment]
Which cache are you speaking about? After someone installs the plugin, the config.xml is in place and therefore cacheable.

**Comment by gaillafr** [comment]
This file is used to get data (not really a caching system) without instantiating the module.
If a merchant updates something like 20 modules it will take more time for the shop to regenerate the config.xml for each of them.

File: SDK/ShopItem.php (hunk 2 above)
Comment (devyoustice) [inline SDK/ShopItem.php]: we need base64_encode to pass the image of the product to the API server in JSON. So if you're fine using base64_encode for this purpose, we're fine removing this workaround.

**Comment by devyoustice** [comment]
New version updated. This also removes the work-around for base64_encode.

**Comment by Quetzacoalt91** [comment]
Hi @devyoustice,

Your PR has just been merged and the changes should be available from tomorrow.

Regards

**Comment by devyoustice** [comment]
thanks.

**Pull request merged**